controllers/compaction/reconciler.go

```diff
@@ -384,8 +389,25 @@ func getCompactionJobVolumes(etcd *druidv1alpha1.Etcd, logger logr.Logger) []v1.
 		logger.Error(err, "Storage provider is not recognized. Compaction job will fail as no storage could be configured", "namespace", etcd.Namespace, "name", etcd.Name)
 		return vs
 	}
+	switch provider {
+	case "Local":
+		hostPath, err := utils.GetHostMountPathFromSecretRef(ctx, cl, logger, storeValues, etcd.Namespace)
+		if err != nil {
+			logger.Error(err, "Host mount path could not be detremined from provided secrets", "namespace", etcd.Namespace, "name", etcd.Name)
+			return vs
+		}
 
-	if provider == utils.GCS || provider == utils.S3 || provider == utils.OSS || provider == utils.ABS || provider == utils.Swift || provider == utils.OCS {
+		hpt := v1.HostPathDirectory
+		vs = append(vs, v1.Volume{
+			Name: "host-storage",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: hostPath + "/" + pointer.StringDeref(storeValues.Container, ""),
```

Comment:
```suggestion
					Path: path.Join(hostPath, *storeValues.Container),
```

Reply:
What's wrong here? `StringDeref` checks nil poiinter error as well

Reply:
If it's possible to avoid a function call with such simple syntax for pointer dereference, then why not use it?

Reply:
This function call is doing a nil pointer check as well. I would prefer to persist with it

Reply:
Ok. But please use path.Join(), since that's the safe and recommended way of generating path strings, especially if we intend to run druid on a different platform/architecture tomorrow.